### PR TITLE
planner: consider using instance level plan cache 

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: ref #54057

Problem Summary: planner: consider using instance level plan cache 

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]

File: pkg/planner/core/plan_cache_utils.go [hunk 1]
```diff
@@ -291,7 +291,6 @@ func NewPlanCacheKey(sctx sessionctx.Context, stmt *PlanCacheStmt) (key, binding
 
 	hash := make([]byte, 0, len(stmt.StmtText)*2) // TODO: a Pool for this
 	hash = append(hash, hack.Slice(stmtDB)...)
-	hash = codec.EncodeInt(hash, int64(vars.ConnectionID))
```
Comment (AilinKid) [inline pkg/planner/core/plan_cache_utils.go:294]: even the session level plan doesn't encode connection id any more?

**Review by AilinKid — COMMENTED** [review]
Rest LGTM

File: pkg/planner/core/plan_cache_utils.go (hunk 1 above)
Comment (qw4990) [reply pkg/planner/core/plan_cache_utils.go:294]: Yes, each session has its own separated Session Plan Cache, so this connectionID is unnecessary in this case as well.

**Pull request merged**